### DISCUSSION TITLE
[gatsby-source-contentful] getaddrinfo ENOTFOUND cdn.contentful.com with corporate proxy

### DISCUSSION THREAD

**Discussion by mwskwong** [gatsbyjs/gatsby#33497] [Help]
I've tried everything. I'm tired and I desperately need help.

I'm working on the same project with `gatsby-source-contentful` installed both at the office and at home. When at home, everything works fine. But at office, the project won't even start because it can't reach the contentful CDN.
```
FetchError: request to https://cdn.contentful.com/spaces/zz9cwhc5t97i/environments/master/content_types failed, 
reason: getaddrinfo ENOTFOUND cdn.contentful.com (value)
```

I know our office has a corporate proxy installed so I tend to believe it is the proxy's fault. In fact, I can't even ping `cdn.contentful.com` directly because of that. Yet I have tried multiple ways of setting the proxy and none of that works.

**Does anyone have some suggestions? Or perhaps it is not even the proxy's fault?**

Things I have tried
1. Setting the `proxy` option in `gatsby-config.js` as follow:
Yes, I made sure those `HA_*` env variables are 100% correct.
```js
{
  resolve: "gatsby-source-contentful",
  options: {
    proxy: HA_PROXY_HOST && {
      host: HA_PROXY_HOST,
      port: HA_PROXY_PORT,
      auth: {
        username: HA_PROXY_USER,
        password: HA_PROXY_PASSWORD
      }
    }
  }
}
```

2. Similar to (1), but with specifying the `protocol` option. For some reason, `gatsby-source-contentful` reports that it doesn't support it despite `axios` does.
```js
{
  resolve: "gatsby-source-contentful",
  options: {
    proxy: HA_PROXY_HOST && {
      protocol: "http",
      host: HA_PROXY_HOST,
      port: HA_PROXY_PORT,
      auth: {
        username: HA_PROXY_USER,
        password: HA_PROXY_PASSWORD
      }
    }
  }
}
```

3. Setting `http_proxy` and `https_proxy` env variables, as suggested by `axios`.

4. Messing around in the Windows proxy setting

**Comment by KyleAMathews** [comment]
I'm pretty sure the Axios proxy is for something else.

You really need to ask your company to allow access to contentful.com. Perhaps you can hotspot off your phone in the meantime?

You could also use a VPN to get around the corporate firewall https://www.pcmag.com/picks/the-best-vpn-services

**Reply by mwskwong** [reply]
> I'm pretty sure the Axios proxy is for something else.
> 
> You really need to ask your company to allow access to contentful.com. Perhaps you can hotspot off your phone in the meantime?
> 
> You could also use a VPN to get around the corporate firewall https://www.pcmag.com/picks/the-best-vpn-services

Damn...
So there is nothing I can do on my end besides either (1) ask the network team to let me through (which is not likely to happen) or (2) workaround/bypass the proxy (the security team will not be happy with that).

**Reply by mwskwong** [reply]
I should also mention that I do have auth access to the proxy. That means as soon as I provide the right username and password of the proxy, Gatsby, npm, and Node.js do have access to the internet. The issue is, whether there is a way for me to specify that information.

Will that change anything?
